crypto/sr25519/pubkey.go

```diff
@@ -0,0 +1,71 @@
+package sr25519
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/tendermint/tendermint/crypto"
+	"github.com/tendermint/tendermint/crypto/tmhash"
+
+	schnorrkel "github.com/ChainSafe/go-schnorrkel"
+)
+
+var _ crypto.PubKey = PubKeySr25519{}
+
+// PubKeySr25519Size is the number of bytes in an Sr25519 public key.
+const PubKeySr25519Size = 32
+
+// PubKeySr25519 implements crypto.PubKey for the Sr25519 signature scheme.
+type PubKeySr25519 [PubKeySr25519Size]byte
```

Comment:
Should be unsized. 

I understand this is a copy of what is in Tendermint but maybe the sdk should first start with migrating what it uses (secp256k1). I understand I said we should migrate the other keys but if the other keys are not used then it is better to start with the minimum (secp256k1) and build up from there? I am up for a discussion on this. 

I am proposing this because when I switched sims to ed25519 keys they failed. This makes me think there is a reliance on key type currently?

Reply:
thoughts @aaronc @alexanderbez

Reply:
Yeah we can start with just SECP256K1 for now. Also, they should be unsized if we want to be able to use `oneof` I think?

Reply:
We do want unsized bye arrays, is it okay if we do that in a separate PR? The intention here was mainly to do an exact copy so we don't have to get both the copying and the changes at once

Reply:
Should the copy be of master? Much of these changes are already made.

Reply:
Okay, we'll need to update this against master then.